keras/backend/common.py

```diff
@@ -145,6 +145,29 @@ def set_image_data_format(data_format):
     _IMAGE_DATA_FORMAT = str(data_format)
 
 
+class BackendFunction(object):
```

Comment:
This let's you add backend specific code with specialized functionality without affecting any cross-backend code used to train and predict. 

For instance if you build a TF op that logs image data from a reinforcement learning image source, you could run that through this mechanism.